### PR TITLE
PC-1293: Integrity check (TermExpression)

### PR DESCRIPTION
**Pull request by ulfryk**

### CODE REVIEW

File: src/ast/expression.spec.ts [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+          returnType: ValueType.Text,
+          type: ExpressionType.Text,
+          value: ' lorem ipsum ',
+        },
+        type: ExpressionType.Phrase,
+        value: ' lorem ipsum ',
+      },
+    ];
+
+    // {
```
Comment (tomecko) [inline src/ast/expression.spec.ts]: when will it be uncommented?

File: src/ast/expression.spec.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+    const invalidDTO = [
+
+      // --- TEXT ---
+      [{
+        preparedValue: ' c c c ',
+        returnType: ValueType.Text,
+        type: ExpressionType.Text,
+        value: 'ccc',
+      }, ['Values of TextExpression doesn\'t match: { value: ccc, preparedValue:  c c c }']],
```
Comment (tomecko) [inline src/ast/expression.spec.ts]: Values don't

File: src/ast/expression.spec.ts [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        type: ExpressionType.Selector,
+        value: 'first_name',
+      }, ['SelectorExpression is invalid - "matchingType" should equal TEXT but is NUMBER']],
+      [{
+        matchingType: ValueType.Text,
+        preparedValue: 'firstName',
+        returnType: ValueType.Text,
+        type: ExpressionType.Selector,
+        value: 'firstName',
+      }, ['SelectorExpression is invalid firstName is not available on model definition']],
```
Comment (tomecko) [inline src/ast/expression.spec.ts]: hyphen after "invalid"?

File: src/ast/expression.spec.ts [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+import { fromJS } from './from-js';
+
+const getErrors = (e: Expression) => e.toList()
+  .filter(node => node instanceof InvalidExpression)
+  .flatMap(({ errors }: InvalidExpression) => errors)
+  .toArray();
+
+describe('SearchQL ast: Expression tree', () => {
+
+  xit('should be able to self check types', () => {
```
[the remark this answers is not in the document]
Comment (ulfryk) [reply src/ast/expression.spec.ts]: yep, `type check` tests will be added when `integrityCheck` feature is finished across all kinds of AST nodes

File: src/ast/expression.spec.ts (hunk 1 above)
Comment (ulfryk) [reply src/ast/expression.spec.ts]: step by step while adding tests for next nodes

**Pull request merged**